foundation_cms/legacy_apps/wagtailpages/pagemodels/base.py

```diff
@@ -35,14 +38,21 @@
 )
 
 
-class BasePage(FoundationMetadataPageMixin, FoundationNavigationPageMixin, Page):
+class BasePage(FoundationNavigationPageMixin, FoundationMetadataPageMixin, Page):
     class Meta:
         abstract = True
 
     def get_donate_banner(self, request):
+        # Decide legacy vs redesign from the FoundationNavigationPageMixin
+        parent_homepage = getattr(request, "parent_homepage", None)
+
         # Check if there's a DonateBannerPage.
         default_locale = Locale.get_default()
-        donate_banner_page = DonateBannerPage.objects.filter(locale=default_locale).first()
+
+        if parent_homepage == "redesign":
+            donate_banner_page = SitewideDonateBannerPage.objects.filter(locale=default_locale).first()
+        else:
+            donate_banner_page = DonateBannerPage.objects.filter(locale=default_locale).first()
```

Comment:
Since we introduced the `SiteWideDonateBannerPage`, this legacy `BasePage` method has been updated to check whether or not the page lives under the redesign homepage. If it does, we want to check for the the new `SiteWideDonateBannerPage`, if not, we want the legacy `DonateBannerPage` (though that scenario is very unlikely. No pages live under the legacy homepage anymore).